lib/IonRequest.ts

```diff
@@ -265,6 +315,16 @@ export default class IonRequest {
     if (publicKeyJwk.y.length !== 43) {
       throw new IonError(ErrorCode.JwkEs256kHasIncorrectLengthOfY, `SECP256K1 JWK 'y' property must be 43 bytes.`);
     }
+
+    if (isPrivateKey && (publicKeyJwk.d === undefined || publicKeyJwk.d.length !== 43)) {
+      throw new IonError(ErrorCode.JwkEs256kHasIncorrectLengthOfY, `SECP256K1 JWK 'd' property must be 43 bytes.`);
```

Comment:
```suggestion
      throw new IonError(ErrorCode.JwkEs256kHasIncorrectLengthOfD, `SECP256K1 JWK 'd' property must be 43 bytes.`);
```
Also technically, it is optional to check for schema for private key because:
1. It is only used internally in the SDK.
2. Surely an error would be thrown when the code is trying to sign using a public key later.